### ISSUE TITLE
#include <format> 

### ISSUE THREAD

**Issue by sam-mix** [Apress/beginning-cpp20#42]
1. env:

```bash
gcc --version
gcc (GCC) 11.3.1 20221121 (Red Hat 11.3.1-4)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


g++ --version
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: arm64-apple-darwin22.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

```

2. error:
```
cd "/root/ws/beginning-cpp20/Examples/NoModules/Chapter 02/" && g++ -std=c++20 Ex2_03B.cpp -o Ex2_03B.exe && "/root/ws/beginning-cpp20/Examples/NoModules/Chapter 02/"Ex2_03B.exe
Ex2_03B.cpp:3:10: fatal error: format: No such file or directory
    3 | #include <format>
      |          ^~~~~~~~
compilation terminated.

```

3. cppreference

[cppreference](https://en.cppreference.com/w/cpp/utility/format/format)

4. my opinion

It's like a joke

**Comment by Peter-VanWeert** [comment]
I understand your frustration, but there's is very little we can do about this. 

When you update your book for a new language standard, you can only hope that compiler and library writers catch up in a timely manner (and they usually do). To be honest, I never expected the `<format>` header to be a problem, given that there was already a perfectly perfect [reference implementation](https://fmt.dev/) long before C++20 appeared. I was more worried about concepts and modules (and the latter turned out to be a problem indeed...). But for reasons I don't fully understand, it did unfortunately take years for most libraries to add support for `<format>`...

Either way: I'm not sure what you expect from me at this point. 

First, the markdown on both the Examples and Exercises sections clearly refer to the  [Workarounds](https://github.com/Apress/beginning-cpp20/blob/main/Workarounds) section where we acknowledge this and other challenges like it, and give you pointers on how you can work around it. (Added one more to the main page just now, so it'll be even harder to miss...) Even at the time we wrote the book, if I recall, you could compile nearly all examples with all major C++20 era compilers using these workarounds.

> Consult the [Workarounds](https://github.com/Apress/beginning-cpp20/blob/main/Workarounds) directory on how to work around compilation issues with other C++20 language and library features that your compiler may not (fully) [support](https://en.cppreference.com/w/cpp/compiler_support) yet.

Second, afaik, the latest versions of all major libraries now (finally) have native support for `<format>`. I've moved on, and am now crossing my fingers the do the same for `<print>` by the time we publish Beginning C++23.

So, in conclusion, thanks for sharing this lovely joke with us, but I'm going to consider this issue closed...
